apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/context/hooks.ts

```diff
@@ -0,0 +1,196 @@
+import { useEffect } from "react";
+import { useSelector } from "react-redux";
+
+import { TransactionAnalyzer } from "@caravan/fees";
+
+import { FeePriority, useFeeEstimates } from "clients/fees";
+import { usePendingUtxos, useWalletUtxos } from "hooks/utxos";
+import {
+  selectWalletConfig,
+  getExtendedPublicKeyImporters,
+  getWalletAddresses,
+  getChangeAddresses,
+} from "selectors/wallet";
+import { MultisigAddressType, Network } from "@caravan/bitcoin";
+
+import {
+  setFeeBumpStatus,
+  setFeeBumpError,
+  setFeeBumpRecommendation,
+} from "./feeBumpActions";
+
+import { FeeBumpStatus, FeeBumpRecommendation } from "../types";
+import { extractUtxosForFeeBumping, validateTransactionInputs } from "../utils";
+import { TransactionDetails } from "@caravan/clients";
+import { useFeeBumpContext } from "./FeeBumpContext";
+// =============================================================================
+// HOOKS
+// =============================================================================
+
+/**
+ * Identifies the change output in a transaction by analyzing output addresses
+ * and wallet data
+ *
+ * This function uses multiple heuristics to identify which output is the change:
+ * 1. Matches against known wallet addresses
+ * 2. Checks BIP32 path patterns (change addresses use path m/1/*)
+ * 3. Position in outputs (change is often the last output)
+ *
+ * @param transaction - The transaction object
+ * @param walletState - The wallet state containing addresses
+ * @returns Index of the change output or undefined if not found
+ *
+ * @see https://en.bitcoin.it/wiki/Privacy#Change_address_detection
+ */
+export const useChangeOutputIndex = (
+  transaction?: TransactionDetails,
```

Comment:
So, this hook should get re-called if the tx value changes and something new gets passed through. then this hook will return a new index value. So I think this works as a custom hook. We'll have to see though.

Reply:
It'll only run once unless the the transaction object changes.